### PR TITLE
Refactor to use MapLibreUtils from an index as a local library

### PR DESCRIPTION
**Pull request by Pete-Y-CS**
this was painful and I now see can't automatically merge
Will rebase tomorrow

### CODE REVIEW

File: src/lib/layers/SpeedLimits.svelte
Comment (dabreegster) [inline src/lib/layers/SpeedLimits.svelte:1]: Hmm, this is a huge change, I meant we still import individual functions, but from "../maplibre". We'd have maplibre/index.ts that does stuff to re-export things from multiple files, like maplibre/utils.ts and maplibre/layers_helpers.ts. We can have this "singleton with a bunch of static methods" pattern (not the right term, that'd be Java) if it seems nice or we don't want to verbosely import individual methods, if you like it

File: src/lib/maplibre/maplibre_utils.ts [hunk 1]
```diff
@@ -21,16 +21,16 @@ const defaultColor = "#000000";
 const defaultFilter = true;
 const defaultOpacity = 1;
 
-export const isPolygon: FilterSpecification = ["==", "$type", "Polygon"];
-export const isLine: FilterSpecification = ["==", "$type", "LineString"];
-export const isPoint: FilterSpecification = ["==", "$type", "Point"];
+const isPolygon: FilterSpecification = ["==", "$type", "Polygon"];
```
Comment (dabreegster) [inline src/lib/maplibre/maplibre_utils.ts:24]: I think we'd keep the exports here

File: src/lib/maplibre/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export { default as MapLibreUtils } from "./maplibre_utils";
```
Comment (dabreegster) [inline src/lib/maplibre/index.ts:1]: I think something like `export * from "./utils"` should work. If it doesn't, we have to do something like `export { isPolygon, isLine, isPoint, ... } from "./utils"`. Can generate that list pretty quickly with grep and a regex if you like

File: src/lib/layers/SpeedLimits.svelte
Comment (dabreegster) [reply src/lib/layers/SpeedLimits.svelte:1]: Up to you, whatever's easiest -- guessing there's a slight sunk cost here if vscode or regexes wasn't the secret behind this PR so far ;)

Comment (Pete-Y-CS) [reply src/lib/layers/SpeedLimits.svelte:1]: There's a feeling of sunk cost but I just tried `export * from "./utils"` anyway and this didn't work as a first pass. If you're enough of a regex master to furnish me with a list quickly that would be ace

Comment (Pete-Y-CS) [reply src/lib/layers/SpeedLimits.svelte:1]: unlesss... give me a mo I might be wrong there

Comment (dabreegster) [reply src/lib/layers/SpeedLimits.svelte:1]: export { isPolygon, isLine, isPoint, overwriteSource, overwritePmtilesSource, overwriteLayer, overwritePolygonLayer, overwriteCircleLayer, overwriteLineLayer, emptyGeojson, pointFeature, setPrecision, constructMatchExpression, hoveredToggle, makeColorRamp, bbox, prettyPrintMeters, type FeatureWithProps } from "./utils"

**Pull request closed without merging**